### ISSUE TITLE
DeprecationWarning: Deprecated since Odoo 18: ustr()

### ISSUE THREAD

**Issue by chrisb-c01** [OCA/openupgradelib#393]
I am running a `logged_query()` on openupgradelib==3.7.1 and getting a deprecation warning:

```
2024-12-10 15:33:42,390 702 WARNING odoo py.warnings: /opt/odoo-venv/lib/python3.10/site-packages/openupgradelib/openupgrade.py:1724: DeprecationWarning: Deprecated since Odoo 18: ustr() is a garbage bag of weirdo fallbacks which mostly don't do anything as
- the first attempt will always work if errors is not `strict`
- if utf8 fails it moves on to latin-1 which always works
- and it always tries hint-encoding twice
  File "/opt/odoo-venv/bin/coverage", line 8, in <module>
    sys.exit(main())
  File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/cmdline.py", line 970, in main
    status = CoverageScript().command_line(argv)
  File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/cmdline.py", line 681, in command_line
    return self.do_run(options, args)
  File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/cmdline.py", line 858, in do_run
    runner.run()
  File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/execfile.py", line 208, in run
    exec(code, main_mod.__dict__)
  File "/opt/odoo-venv/bin/odoo", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/opt/odoo/setup/odoo", line 8, in <module>
    odoo.cli.main()
  File "/opt/odoo/odoo/cli/command.py", line 66, in main
    o.run(args)
  File "/opt/odoo/odoo/cli/server.py", line 180, in run
    main(args)
  File "/opt/odoo/odoo/cli/server.py", line 173, in main
    rc = odoo.service.server.start(preload=preload, stop=stop)
  File "/opt/odoo/odoo/service/server.py", line 1402, in start
    rc = server.run(preload, stop)
  File "/opt/odoo/odoo/service/server.py", line 577, in run
    rc = preload_registries(preload)
  File "/opt/odoo/odoo/service/server.py", line 1306, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "<decorator-gen-13>", line 2, in new
  File "/opt/odoo/odoo/tools/func.py", line 97, in locked
    return func(inst, *args, **kwargs)
  File "/opt/odoo/odoo/modules/registry.py", line 127, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/opt/odoo/odoo/modules/loading.py", line 480, in load_modules
    processed_modules += load_marked_modules(env, graph,
  File "/opt/odoo/odoo/modules/loading.py", line 364, in load_marked_modules
    loaded, processed = load_module_graph(
  File "/opt/odoo/odoo/modules/loading.py", line 192, in load_module_graph
    getattr(py_module, pre_init)(env)
  File "/__w/odoo-project/odoo-project/some_module/__init__.py", line 11, in pre_init_hook
    openupgrade.logged_query(
  File "/opt/odoo-venv/lib/python3.10/site-packages/openupgradelib/openupgrade.py", line 1724, in logged_query
    full_query = tools.ustr(cr._obj.query)
```

**Comment by StefanRijnhart** [comment]
Fixed here: https://github.com/OCA/openupgradelib/pull/386, but new releases have to be created manually. I just released 3.7.2, which has become available on Pypi. Can you try upgrading?

**Comment by chrisb-c01** [comment]
Thanks @StefanRijnhart much appreciated.
